wg-device-management/charter.md

```diff
@@ -0,0 +1,100 @@
+  those infrastructure components.
+- Enable in-node devices as well as network-accessible devices.
+- Minimize workload disruption due to hardware failures.
+- Address fragmentation of accelerator due to fractional use.
+- Additional problems that may be identified and deemed in scope as we gather
+  use cases and requirements from WG Serving, WG Batch, and other stakeholders.
+- Address all of the above while with a simple API that is a natural extension
+  of the existing Kubernetes APIs, and avoids or minimizes any transition
+  effort.
+
+### Out of Scope
+
+- Higher-level workload controller APIs (for example, the equivalent of
+  Deployment, StatefulSet, or DaemonSet) for specific types of workloads.
+- General resource management requirements not related to devices.
+
+## Deliverables
+
+The WG will coordinate the delivery of KEPs and their implementations by the
+participating SIGs. Interim artifacts will include documents capturing use
+cases, requirements, and designs; however, all of those will eventually result
+in KEPs and code owned by SIGs.
+
+Specifically, we expect to need:
+
+- APIs for publishing resource capacity of in-node and network-accessible
+  devices, as well as sample code to ease creation of drivers to populate this
+  information.
+- APIs for specifying workload resource requirements with respect to devices.
+- APIs, algorithms, and implementations for allocating access to and resources on devices, as well as
+  persisting the results of those allocations.
+- APIs, algorithms, and implementations for allowing adminstrators to control
+  and govern access to devices.
+
+## Stakeholders
+
+- SIG Architecture
+- SIG Autoscaling
+- SIG Network
+- SIG Node
+- SIG Scheduling
+
+Additionally a broad set of end users, device vendors, cloud providers,
+Kubernetes distribution providers, and ecosystem projects (particularly
+autoscaling-related projects) have expressed interest in this effort. There are
+five primary groups of stakeholders from each of which we expect multiple participants:
+
+- Device vendors that manufacture accelerators and other specialized hardware
+  which they would like to make available to Kubernetes users.
+- Kubernetes distribution and managed offering providers that would like to make
+  specialized hardware available to their users.
+- Kubernetes ecosystem projects that help manage workloads utilizing these
+  accelerators (e.g., Karpenter, Kueue, Volcano)
+- End user workload authors that will create workloads that take advantage of
+  the specialized hardware.
+- Cluster administrators that operate and govern clusters containing the
+  specialized hardware.
+
+## Roles and Organization Management
+
+This working group adheres to the Roles and Organization Management outlined in
+[wg-governance] and opts-in to updates and modifications to [wg-governance].
+
+## Exit Criteria
+
+The working group will disband when the KEPs resulting from these discussions
+have reached a terminal state.
```

Comment:
This seems a bit vague. 

I could see that wg-device-management should go away once DRA has reached GA (and maybe 2-3 vendors) are widely supported with DRA in GA.

To be clear, this does not mean work is done in this area but I think the urgency of this group will go down once DRA is out and supported in many places.

Reply:
Maybe? I think saying "DRA reaches GA" is also vague. The whole of the functionality will be delivered incrementally, most likely. So there isn't one or two existing KEPs that will go to GA and we're done. I expect a few more things to come out of this.

Reply:
Yea, I'm not sure of a clear exit criteria then. In serving it was asked to come up with clear goals so we know when to call the working group complete.

Reply:
ref: https://github.com/kubernetes/community/pull/7823/files#r1581095409

Reply:
I added a note saying we would consider disbanding when core DRA reaches GA.